docs/guides/migrating-to-async-api.mdx

```diff
@@ -0,0 +1,185 @@
+const store = createStore()
+store.set(fooAtom, 'foo')
+```
+
+You can also create your own React Context to pass a store.
+
+## What are breaking
+
+### Import statements
+
+The new API is provided from different entry points:
+
+- `jotai/vanilla`
+- `jotai/react`
+- `jotai/react/devtools`
+- `jotai/react/utils`
+
+```js
+import { atom } from 'jotai/vanilla'
+import { useAtom } from 'jotai/react'
+```
+
+These new entry points are added in v1.10.0.
+
+In v2.0.0, they are the defaults and old entry points simply refer to the new ones.
+
+```js
+import { atom } from 'jotai' // is same as 'jotai/vanilla'
+import { useAtom } from 'jotai' // is same as 'jotai/react'
+```
+
+### Async atoms are no longer special
+
+Async atoms are just normal atoms with promise values.
+Atoms getter functions don't resolve promises.
+On the other hand, `useAtom` hook continues to resolve promises.
+
+### Some functions are dropped
+
+- Provider's `initialValues` prop is removed, because `store` is more flexible.
+- Provider's scope props is removed, because you can create own context.
+- `abortableAtom` util is removed, becuase the feature is included by default
+- `waitForAll` util is removed, because `Promise.all` just works
+
+## Migration guides
+
+### Async atoms
+
+`get` function for read function of async atoms
+doesn't resolve promises, so you have to put `await`.
+
+In short, the change is something like the following.
+(If you are TypeScript users, types will tell where to changes.)
+
+#### Previous API
+
+```js
+const asyncAtom = atom(async () => 'hello')
+const derivedAtom = atom((get) => get(asyncAtom).toUppercase())
+```
+
+#### New API
+
+```js
+const asyncAtom = atom(async () => 'hello')
+const derivedAtom = atom(async (get) => (await get(asyncAtom)).toUppercase())
+```
+
+TODO: provide `unwrap` util
+
+### Provider's `initialValues` prop
+
+#### Previous API
+
+```jsx
+const countAtom = atom(0)
+
+  <Provider initialValues={[[countAtom, 1]]}>
+    ...
+  </Provider>
+```
+
+#### New API
+
+```jsx
+const countAtom = atom(0)
+const store = createStore()
+store.set(countAtom, 1)
+
+  <Provider store={store}>
+    ...
+  </Provider>
+```
+
+### Provider's `scope` prop
+
+#### Previous API
+
+```jsx
+const myScope = Symbol()
+
+  // Parent component
+  <Provider scope={myScope}>
+    ...
+  </Provider>
+
+  // Child component
+  useAtom(..., myScope)
+```
+
+#### New API
+
+```jsx
+const MyContext = createContext()
+const store = createStore()
+
+  // Parent component
+  <MyContext.Provider store={store}>
+    ...
+  </MyContext.Provider>
+
+  // Child Component
+  const store = useContext(MyContext)
+  useAtom(..., { store })
+```
+
+### `abortableAtom` util
+
+#### Previous API
+
+```js
+const asyncAtom = abortableAtom(async (get, { signal }) => {
+ ...
+}
+```
+
+#### New API
+
+```js
+const asyncAtom = atom(async (get, { signal }) => {
+  ...
+}
+```
+
+### `waitForAll` util
+
+#### Previous API
+
+```js
+const allAtom = waitForAll([fooAtom, barAtom])
```

Comment:
Maybe you should describe that waitForAll is not needed, as it can be handled by native Promise APIs.